src/osd/OSD.cc

```diff
@@ -7936,6 +7936,18 @@ void OSD::consume_map()
   assert(osd_lock.is_locked());
   dout(7) << "consume_map version " << osdmap->get_epoch() << dendl;
 
+  /** make sure the cluster is speaking in SORTBITWISE, because we don't
+   *  speak the older sorting version any more. Be careful not to force
+   *  a shutdown if we are merely processing old maps, though.
+   */
+  if (!osdmap->test_flag(CEPH_OSDMAP_SORTBITWISE)) {
+    epoch_t boot = service.get_boot_epoch();
+    if (boot && boot <= osdmap->get_epoch()) {
```

Comment:
Can we just condition this on is_active() instead?  I think the piece that matters is us communicating iwth other OSDs, which is controlled here:
<pre>
void OSD::dispatch_context(PG::RecoveryCtx &ctx, PG *pg, OSDMapRef curmap,
                           ThreadPool::TPHandle *handle)
{
  if (service.get_osdmap()->is_up(whoami) &&
      is_active()) {
    do_notifies(*ctx.notify_list, curmap);
    do_queries(*ctx.query_map, curmap);
    do_infos(*ctx.info_map, curmap);
  }
</pre>
..and that is conditioned on is_active() (and is_up()... although i'm not quite sure when that diverges from is_active()).

I don't think boot epoch is reset to 0 if we get marked down, which means we would abort trying to chew through maps while we were down if the flag is set then, whereas i think we only should crash if we get marked up and it is set?

Reply:
I don't think the cases where we are marked down and don't update boot_epoch will matter — if sortbitwise gets turned *off* we want to complain; and if it gets turned back on we'll happily churn through when restarted — but I can make that change. There was some reason not to when doing it in the PG case but it works here.

(We have the double-check in that condition because we can prepublish maps before doing enough processing on them to make sure we're still supposed to be active.)